lib/instrumentation/bunyan.js

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2022 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const {
+  isApplicationLoggingEnabled,
+  isLogForwardingEnabled,
+  isLocalDecoratingEnabled,
+  isMetricsEnabled,
+  createModuleUsageMetric,
+  incrementLoggingLinesMetrics
+} = require('../util/application-logging')
+
+const MAX_LENGTH = 1021
+const OUTPUT_LENGTH = 1024
+const truncate = (str) => {
+  if (str && str.length > OUTPUT_LENGTH) {
+    return str.substring(0, MAX_LENGTH) + '...'
+  }
+
+  return str
+}
+
+const logger = require('../logger').child({ component: 'bunyan' })
+
+module.exports = function instrument(agent, bunyan, _, shim) {
+  const config = agent.config
+
+  if (!isApplicationLoggingEnabled(config)) {
+    logger.debug('Application logging not enabled. Not instrumenting bunyan.')
+    return
+  }
+
+  const logForwardingEnabled = isLogForwardingEnabled(config, agent)
+  const metricsEnabled = isMetricsEnabled(config)
+
+  if (logForwardingEnabled || metricsEnabled) {
+    shim.wrap(bunyan, 'createLogger', function wrapCreateLogger(_shim, createLogger) {
+      return function wrappedCreateLogger() {
+        createModuleUsageMetric('bunyan', agent.metrics)
+
+        const bunyanLogger = createLogger.apply(this, arguments)
+
+        if (logForwardingEnabled) {
+          bunyanLogger.addStream({
+            type: 'raw',
+            level: bunyanLogger.level(),
+            stream: {
+              write: function nrLogWrite(logLine) {
+                agent.logs.add(logLine)
+              }
+            }
+          })
+        }
+        return bunyanLogger
+      }
+    })
+  }
+
+  const localDecoratingEnabled = isLocalDecoratingEnabled(config)
+
+  shim.wrap(bunyan.prototype, '_emit', function wrapEmit(shim, emit) {
```

Comment:
I think this instrumentation should only count log lines when metrics are enabled and append the piped nr linking metadata when local log decorating is enabled.  The rest should go in the stream write handler above